# On branch m-100
# Changes to be committed:
#   (use "git reset HEAD <file>..." to unstage)
#
#	modified:   base-files/etc/uci-defaults/01_leds
#	modified:   base-files/lib/ramips.sh
#	modified:   base-files/lib/upgrade/platform.sh
#	new file:   dts/M-100.dts
#	modified:   image/Makefile
#	new file:   mt7620a/profiles/baidu.mk
diff --git a/target/linux/ramips/base-files/etc/uci-defaults/01_leds b/target/linux/ramips/base-files/etc/uci-defaults/01_leds
index 567835e..c9c1d4b 100755
--- a/target/linux/ramips/base-files/etc/uci-defaults/01_leds
+++ b/target/linux/ramips/base-files/etc/uci-defaults/01_leds
@@ -103,6 +103,8 @@ case $board in
 	hw550-3g)
 		set_usb_led "hw550-3g:green:usb"
 		;;
+	m-100)
+		;;
 	mofi3500-3gn)
 		set_usb_led "mofi3500-3gn:green:usb"
 		;;
diff --git a/target/linux/ramips/base-files/lib/ramips.sh b/target/linux/ramips/base-files/lib/ramips.sh
index a3a26cb..57ef110 100755
--- a/target/linux/ramips/base-files/lib/ramips.sh
+++ b/target/linux/ramips/base-files/lib/ramips.sh
@@ -64,6 +64,9 @@ ramips_board_detect() {
 	*"Aztech HW550-3G")
 		name="hw550-3g"
 		;;
+	*"BAIDU M-100")
+		name="m-100"
+		;;
 	*"DCS-930")
 		name="dcs-930"
 		;;
diff --git a/target/linux/ramips/base-files/lib/upgrade/platform.sh b/target/linux/ramips/base-files/lib/upgrade/platform.sh
index 8ff6063..f8c8672 100755
--- a/target/linux/ramips/base-files/lib/upgrade/platform.sh
+++ b/target/linux/ramips/base-files/lib/upgrade/platform.sh
@@ -50,6 +50,7 @@ platform_check_image() {
 	hg255d | \
 	hlk-rm04 | \
 	ip2202 | \
+	m-100 | \
 	m3 | \
 	m4 | \
 	mofi3500-3gn | \
diff --git a/target/linux/ramips/dts/M-100.dts b/target/linux/ramips/dts/M-100.dts
new file mode 100644
index 0000000..99987f6
--- /dev/null
+++ b/target/linux/ramips/dts/M-100.dts
@@ -0,0 +1,146 @@
+/dts-v1/;
+
+/include/ "mt7620a.dtsi"
+
+/ {
+	compatible = "M-100", "ralink,mt7620a-soc";
+	model = "BAIDU M-100";
+
+	palmbus@10000000 {
+		gpio1: gpio@638 {
+			status = "okay";
+		};
+
+		i2c@900 {
+			status = "okay";
+		};
+
+		i2s@a00 {
+			status = "okay";
+
+			pinctrl-names = "default";
+			pinctrl-0 = <&pcm_i2s_pins>;
+		};
+
+		spi@b00 {
+			status = "okay";
+
+			m25p80@0 {
+				#address-cells = <1>;
+				#size-cells = <1>;
+				compatible = "en25q64";
+				reg = <0 0>;
+				linux,modalias = "m25p80", "en25q64";
+				spi-max-frequency = <10000000>;
+
+				partition@0 {
+					label = "u-boot";
+					reg = <0x0 0x30000>;
+					read-only;
+				};
+
+				partition@30000 {
+					label = "u-boot-env";
+					reg = <0x30000 0x10000>;
+					read-only;
+				};
+
+				factory: partition@40000 {
+					label = "factory";
+					reg = <0x40000 0x10000>;
+					read-only;
+				};
+
+				partition@50000 {
+					label = "firmware";
+					// 8M
+					reg = <0x50000 0x7b0000>;
+					// 16M
+					// reg = <0x50000 0xfb0000>; 
+				};
+			};
+		};
+
+		pcm@2000 {
+			status = "okay";
+		};
+
+		gdma@2800 {
+			status = "okay";
+		};
+
+	};
+
+	pinctrl {
+		state_default: pinctrl0 {
+			gpio {
+				ralink,group = "mdio", "rgmii1";
+				ralink,function = "gpio";
+			};
+		};
+	};
+
+	ethernet@10100000 {
+		status = "okay";
+
+		pinctrl-names = "default";
+		pinctrl-0 = <&rgmii1_pins &rgmii2_pins &mdio_pins>;
+
+		ralink,port-map = "llllw";
+
+		port@4 {
+			status = "okay";
+			phy-mode = "rgmii";
+			phy-handle = <&phy4>;
+		};
+
+		port@5 {
+			status = "okay";
+			phy-mode = "rgmii";
+			phy-handle = <&phy5>;
+		};
+
+		mdio-bus {
+			status = "okay";
+
+			phy4: ethernet-phy@4 {
+				reg = <4>;
+				phy-mode = "rgmii";
+			};
+
+			phy5: ethernet-phy@5 {
+				reg = <5>;
+				phy-mode = "rgmii";
+			};
+		};
+	};
+
+	gsw@10110000 {
+		ralink,port4 = "gmac";
+	};
+
+	sdhci@10130000 {
+		status = "okay";
+	};
+
+	pcie@10140000 {
+		status = "okay";
+	};
+
+	gpio-keys-polled {
+		compatible = "gpio-keys";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		poll-interval = <20>;
+		s2 {
+			label = "S2";
+			gpios = <&gpio0 1 1>;
+			linux,code = <0x100>;
+		};
+		s3 {
+			label = "S3";
+			gpios = <&gpio0 2 1>;
+			linux,code = <0x101>;
+		};
+	};
+};
diff --git a/target/linux/ramips/image/Makefile b/target/linux/ramips/image/Makefile
index c6a42ad..d7c1400 100644
--- a/target/linux/ramips/image/Makefile
+++ b/target/linux/ramips/image/Makefile
@@ -661,6 +661,8 @@ Image/Build/Profile/MT7620a_MT7610e=$(call BuildFirmware/Default8M/$(1),$(1),mt7
 Image/Build/Profile/MT7620a_MT7530=$(call BuildFirmware/Default8M/$(1),$(1),mt7620a_mt7530,MT7620a_MT7530)
 Image/Build/Profile/MT7620a_V22SG=$(call BuildFirmware/Default8M/$(1),$(1),mt7620a_v22sg,MT7620a_V22SG)
 Image/Build/Profile/RP-N53=$(call BuildFirmware/Default8M/$(1),$(1),rp_n53,RP-N53)
+Image/Build/Profile/M-100=$(call BuildFirmware/Default8M/$(1),$(1),m-100,M-100)
+#Image/Build/Profile/M-100=$(call BuildFirmware/Default16M/$(1),$(1),m-100,M-100)
 whr_300hp2_mtd_size=7012352
 Image/Build/Profile/WHR300HP2=$(call BuildFirmware/CustomFlash/$(1),$(1),whr-300hp2,WHR-300HP2,$(whr_300hp2_mtd_size))
 Image/Build/Profile/WHR600D=$(call BuildFirmware/CustomFlash/$(1),$(1),whr-600d,WHR-600D,$(whr_300hp2_mtd_size))
@@ -680,6 +682,7 @@ define Image/Build/Profile/Default
 	$(call Image/Build/Profile/WHR300HP2,$(1))
 	$(call Image/Build/Profile/WHR600D,$(1))
 	$(call Image/Build/Profile/NA930,$(1))
+	$(call Image/Build/Profile/M-100,$(1))
 endef
 endif
 
diff --git a/target/linux/ramips/mt7620a/profiles/baidu.mk b/target/linux/ramips/mt7620a/profiles/baidu.mk
new file mode 100644
index 0000000..7521fe4
--- /dev/null
+++ b/target/linux/ramips/mt7620a/profiles/baidu.mk
@@ -0,0 +1,30 @@
+#
+# Copyright (C) 2013 OpenWrt.org
+#
+# This is free software, licensed under the GNU General Public License v2.
+# See /LICENSE for more information.
+#
+
+define Profile/M-100
+	NAME:=BAIDU M-100
+	PACKAGES:=\
+		kmod-usb-core kmod-usb2 kmod-usb-dwc2 \
+		kmod-ledtrig-usbdev \
+		kmod-sound-core kmod-sound-mt7620 \
+		kmod-ledtrig-timer \
+		kmod-usb-acm kmod-usb-net kmod-usb-net-asix kmod-usb-net-rndis kmod-usb-serial kmod-usb-serial-option \
+		kmod-usb-serial-ftdi kmod-usb-serial-pl2303 kmod-usb-serial-mos7720 kmod-usb-serial-ch341 usbutils \
+		usb-modeswitch usb-modeswitch-data comgt \
+		ser2net coreutils coreutils-stty \
+		kmod-usb-storage kmod-usb-storage-extras kmod-fs-ext4 kmod-fs-vfat kmod-fs-ntfs kmod-fuse ntfs-3g block-mount \
+		kmod-nls-cp437 kmod-nls-iso8859-1 block-hotplug \
+		kmod-input-core kmod-soundcore kmod-usb-audio madplay mdnsresponder libspeex mpd mpc \
+		alsa-lib alsa-utils alsa-utils-tests libffmpeg-mini minidlna\
+		luci luci-proto-3g luci-app-minidlna 
+endef
+
+define Profile/M-100/Description
+	Package set for Baidu M-100 Product
+endef
+
+$(eval $(call Profile,M-100))
